compiler/testData/diagnostics/nativeTests/objCRefinement.kt

```diff
@@ -0,0 +1,111 @@
+// FILE: kotlin.kt
+package kotlin.native
+
+@Target(AnnotationTarget.ANNOTATION_CLASS)
+@Retention(AnnotationRetention.BINARY)
+@MustBeDocumented
+annotation class RefinesForObjC
+
+@RefinesForObjC
+@Target(AnnotationTarget.PROPERTY, AnnotationTarget.FUNCTION)
+@Retention(AnnotationRetention.BINARY)
+annotation class RefinedForObjC
+
+@Target(AnnotationTarget.ANNOTATION_CLASS)
+@Retention(AnnotationRetention.BINARY)
+annotation class RefinesInSwift
+
+@RefinesInSwift
+@Target(AnnotationTarget.PROPERTY, AnnotationTarget.FUNCTION)
+@Retention(AnnotationRetention.BINARY)
+public annotation class RefinedInSwift
+
+// FILE: plugin.kt
+package plugin
+
+@RefinesForObjC
+@Target(AnnotationTarget.PROPERTY, AnnotationTarget.FUNCTION)
+@Retention(AnnotationRetention.BINARY)
+annotation class PluginRefinedForObjC
+
+@RefinesInSwift
+@Target(AnnotationTarget.PROPERTY, AnnotationTarget.FUNCTION)
+@Retention(AnnotationRetention.BINARY)
+annotation class PluginRefinedInSwift
+
+// FILE: test.kt
+import plugin.PluginRefinedForObjC
+import plugin.PluginRefinedInSwift
+
+@RefinesForObjC
+<!REDUNDANT_SWIFT_REFINEMENT!>@RefinesInSwift<!>
+@Target(AnnotationTarget.PROPERTY, AnnotationTarget.FUNCTION)
+@Retention(AnnotationRetention.BINARY)
+annotation class MyRefinedAnnotationA
+
+<!INVALID_OBJC_REFINEMENT_TARGETS!>@RefinesForObjC<!>
+@Target(AnnotationTarget.PROPERTY, AnnotationTarget.CLASS)
+@Retention(AnnotationRetention.BINARY)
+annotation class MyRefinedAnnotationB
+
+<!INVALID_OBJC_REFINEMENT_TARGETS!>@RefinesInSwift<!>
+@Retention(AnnotationRetention.BINARY)
+annotation class MyRefinedAnnotationC
```

Comment:
Maybe also check the case when the annotation has only some of the supported targets, e.g. `@Target(AnnotationTarget.FUNCTION)`?

Reply:
Added `MyRefinedAnnotationD` which only has the `PROPERTY` target.